src/app/parcels/form/formSections/CollectionTimeCard.tsx

```diff
@@ -1,35 +1,40 @@
 import React from "react";
-import { onChangeDateOrTime, errorText } from "@/components/Form/formFunctions";
+import { errorText, valueOnChangeDropdownList } from "@/components/Form/formFunctions";
 import GenericFormCard from "@/components/Form/GenericFormCard";
-import { TimePicker } from "@mui/x-date-pickers";
 import { ErrorText } from "@/components/Form/formStyling";
-import dayjs from "dayjs";
 import { ParcelCardProps } from "../ParcelForm";
+import DropdownListInput from "@/components/DataInput/DropdownListInput";
+import { CollectionTimeSlotsLabelsAndValues } from "@/common/fetch";
 
-const CollectionTimeCard: React.FC<ParcelCardProps> = ({
+interface CollectionSlotsCardProps extends ParcelCardProps {
+    collectionTimeSlotsLabelsAndValues: CollectionTimeSlotsLabelsAndValues;
+}
+
+const CollectionSlotCard: React.FC<CollectionSlotsCardProps> = ({
     errorSetter,
     fieldSetter,
     formErrors,
     fields,
+    collectionTimeSlotsLabelsAndValues,
 }) => {
     return (
         <GenericFormCard
-            title="Collection Time"
+            title="Collection Slots"
             required={true}
-            text="What time is the client collecting their parcel?"
+            text="What slot is the client collecting their parcel?"
         >
             <>
-                <TimePicker
-                    onChange={(value) => {
-                        onChangeDateOrTime(fieldSetter, errorSetter, "collectionTime", value);
-                    }}
-                    label="Time"
-                    value={fields.collectionTime ? dayjs(fields.collectionTime) : null}
+                <DropdownListInput
+                    selectLabelId="collection-slot-select-label"
+                    labelsAndValues={collectionTimeSlotsLabelsAndValues}
+                    listTitle="Collection Slot"
+                    defaultValue={fields.collectionSlot ? fields.collectionSlot : ""}
```

Comment:
Is this the bit that sets the slot value when you edit the parcel? Right now if I edit a parcel to select a slot, save, edit again and don't do anything the collection slot is blank. If I press save it changes it to 20:04:50 or 20:04:20 on different parcels - don't really know where it's getting that time from? Maybe the old value from the time picker? Could you fix it to retrieve the correct value please

Reply:
I figured it was because when the parcel info gets passed down the time is not formatted correctly, but sorted now :)